plugin/plugin.go

```diff
@@ -13,13 +20,88 @@ type Args struct {
 	// Level defines the plugin log level.
 	Level string `envconfig:"PLUGIN_LOG_LEVEL"`
 
-	// TODO replace or remove
-	Param1 string `envconfig:"PLUGIN_PARAM1"`
-	Param2 string `envconfig:"PLUGIN_PARAM2"`
+	ApiKey string `envconfig:"PLUGIN_NUGET_APIKEY"`
+	NugetUri string `envconfig:"PLUGIN_NUGET_URI"`
+	PackageLocation string `envconfig:"PLUGIN_PACKAGE_LOCATION"`
 }
 
-// Exec executes the plugin.
-func Exec(ctx context.Context, args Args) error {
-	// write code here
+const globalNugetUri = "https://api.nuget.org/v3/index.json"
+
+func fileExists(name string) bool {
+	if _, err := os.Stat(name); err != nil {
+		if os.IsNotExist(err) {
+			return false
+		}
+	}
+	return true
+}
+
+func validateAndSetArgs(args *Args) error {
+	if args.ApiKey == ""{
+		return fmt.Errorf ("nuget api key must be set in settings")
+	}
+	if args.NugetUri == "" {
+		args.NugetUri = globalNugetUri
+	}
+	if args.PackageLocation != "" && !fileExists(args.PackageLocation){
+		return fmt.Errorf ("the package location: %s does not exist", args.PackageLocation)
+	}
+	return nil
+}
+
+func walkPath(files *[]string) filepath.WalkFunc {
+	return func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			logrus.Errorln(err)
+			return nil
+		}
+		if filepath.Ext(path) == ".nupkg" {
+			*files = append(*files, path)
+		}
+		return nil
+	}
+}
+
+func pushToNuget(file string, args Args) *exec.Cmd {
+	return exec.Command("dotnet", "nuget", "push", file, "--api-key", args.ApiKey, "--source", args.NugetUri, "--skip-duplicate")
+}
+
+func Exec(_ context.Context, args Args) error {
+	logrus.Debug("Starting ...")
+
+	err := validateAndSetArgs(&args)
+	if err != nil {
+		return fmt.Errorf("issues with the parameters passed: %w", err)
+	}
+
+	var files []string
+	// checks if single package location was provided, if not push all.
+	if args.PackageLocation == ""{
+		err = filepath.Walk(".", walkPath(&files))
+		if err != nil {
+			logrus.Errorln(err)
```

Comment:
I recommend the convention `logrus.WithError(err).Errorln("...")

Reply:
Done.